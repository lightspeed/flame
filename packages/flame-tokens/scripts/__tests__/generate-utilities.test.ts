import utilitiesMap from '../generate-utilities';

describe('generate-utilities', () => {
  utilitiesMap.forEach(([fileName, cssOutput]) => {
    describe(`${fileName}.scss`, () => {
      it('should output the right CSS', () => {
        expect(cssOutput).toMatchSnapshot();
      });
    });
  });
});
