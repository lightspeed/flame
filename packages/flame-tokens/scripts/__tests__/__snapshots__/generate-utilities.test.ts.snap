// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generate-utilities _colors.scss should output the right CSS 1`] = `
"@import '../index.scss';
.cr-maple { color: $cr-maple; }
.cr-maple-100 { color: $cr-maple-100; }
.cr-maple-200 { color: $cr-maple-200; }
.cr-maple-300 { color: $cr-maple-300; }
.cr-dive { color: $cr-dive; }
.cr-dive-100 { color: $cr-dive-100; }
.cr-dive-200 { color: $cr-dive-200; }
.cr-dive-300 { color: $cr-dive-300; }
.cr-night { color: $cr-night; }
.cr-night-100 { color: $cr-night-100; }
.cr-night-200 { color: $cr-night-200; }
.cr-night-300 { color: $cr-night-300; }
.cr-snow { color: $cr-snow; }
.cr-snow-100 { color: $cr-snow-100; }
.cr-snow-200 { color: $cr-snow-200; }
.cr-snow-300 { color: $cr-snow-300; }
.cr-blue { color: $cr-blue; }
.cr-blue-100 { color: $cr-blue-100; }
.cr-blue-200 { color: $cr-blue-200; }
.cr-blue-300 { color: $cr-blue-300; }
.cr-green { color: $cr-green; }
.cr-green-100 { color: $cr-green-100; }
.cr-green-200 { color: $cr-green-200; }
.cr-green-300 { color: $cr-green-300; }
.cr-orange { color: $cr-orange; }
.cr-orange-100 { color: $cr-orange-100; }
.cr-orange-200 { color: $cr-orange-200; }
.cr-orange-300 { color: $cr-orange-300; }
.cr-yellow { color: $cr-yellow; }
.cr-yellow-100 { color: $cr-yellow-100; }
.cr-yellow-200 { color: $cr-yellow-200; }
.cr-yellow-300 { color: $cr-yellow-300; }
.cr-gray { color: $cr-gray; }
.cr-gray-100 { color: $cr-gray-100; }
.cr-gray-200 { color: $cr-gray-200; }
.cr-gray-300 { color: $cr-gray-300; }
.cr-seal { color: $cr-seal; }
.cr-seal-100 { color: $cr-seal-100; }
.cr-seal-200 { color: $cr-seal-200; }
.cr-seal-300 { color: $cr-seal-300; }

.cr-bg-maple { background-color: $cr-maple; }
.cr-bg-maple-100 { background-color: $cr-maple-100; }
.cr-bg-maple-200 { background-color: $cr-maple-200; }
.cr-bg-maple-300 { background-color: $cr-maple-300; }
.cr-bg-dive { background-color: $cr-dive; }
.cr-bg-dive-100 { background-color: $cr-dive-100; }
.cr-bg-dive-200 { background-color: $cr-dive-200; }
.cr-bg-dive-300 { background-color: $cr-dive-300; }
.cr-bg-night { background-color: $cr-night; }
.cr-bg-night-100 { background-color: $cr-night-100; }
.cr-bg-night-200 { background-color: $cr-night-200; }
.cr-bg-night-300 { background-color: $cr-night-300; }
.cr-bg-snow { background-color: $cr-snow; }
.cr-bg-snow-100 { background-color: $cr-snow-100; }
.cr-bg-snow-200 { background-color: $cr-snow-200; }
.cr-bg-snow-300 { background-color: $cr-snow-300; }
.cr-bg-blue { background-color: $cr-blue; }
.cr-bg-blue-100 { background-color: $cr-blue-100; }
.cr-bg-blue-200 { background-color: $cr-blue-200; }
.cr-bg-blue-300 { background-color: $cr-blue-300; }
.cr-bg-green { background-color: $cr-green; }
.cr-bg-green-100 { background-color: $cr-green-100; }
.cr-bg-green-200 { background-color: $cr-green-200; }
.cr-bg-green-300 { background-color: $cr-green-300; }
.cr-bg-orange { background-color: $cr-orange; }
.cr-bg-orange-100 { background-color: $cr-orange-100; }
.cr-bg-orange-200 { background-color: $cr-orange-200; }
.cr-bg-orange-300 { background-color: $cr-orange-300; }
.cr-bg-yellow { background-color: $cr-yellow; }
.cr-bg-yellow-100 { background-color: $cr-yellow-100; }
.cr-bg-yellow-200 { background-color: $cr-yellow-200; }
.cr-bg-yellow-300 { background-color: $cr-yellow-300; }
.cr-bg-gray { background-color: $cr-gray; }
.cr-bg-gray-100 { background-color: $cr-gray-100; }
.cr-bg-gray-200 { background-color: $cr-gray-200; }
.cr-bg-gray-300 { background-color: $cr-gray-300; }
.cr-bg-seal { background-color: $cr-seal; }
.cr-bg-seal-100 { background-color: $cr-seal-100; }
.cr-bg-seal-200 { background-color: $cr-seal-200; }
.cr-bg-seal-300 { background-color: $cr-seal-300; }

.cr-border-maple { border-color: $cr-maple; }
.cr-border-maple-100 { border-color: $cr-maple-100; }
.cr-border-maple-200 { border-color: $cr-maple-200; }
.cr-border-maple-300 { border-color: $cr-maple-300; }
.cr-border-dive { border-color: $cr-dive; }
.cr-border-dive-100 { border-color: $cr-dive-100; }
.cr-border-dive-200 { border-color: $cr-dive-200; }
.cr-border-dive-300 { border-color: $cr-dive-300; }
.cr-border-night { border-color: $cr-night; }
.cr-border-night-100 { border-color: $cr-night-100; }
.cr-border-night-200 { border-color: $cr-night-200; }
.cr-border-night-300 { border-color: $cr-night-300; }
.cr-border-snow { border-color: $cr-snow; }
.cr-border-snow-100 { border-color: $cr-snow-100; }
.cr-border-snow-200 { border-color: $cr-snow-200; }
.cr-border-snow-300 { border-color: $cr-snow-300; }
.cr-border-blue { border-color: $cr-blue; }
.cr-border-blue-100 { border-color: $cr-blue-100; }
.cr-border-blue-200 { border-color: $cr-blue-200; }
.cr-border-blue-300 { border-color: $cr-blue-300; }
.cr-border-green { border-color: $cr-green; }
.cr-border-green-100 { border-color: $cr-green-100; }
.cr-border-green-200 { border-color: $cr-green-200; }
.cr-border-green-300 { border-color: $cr-green-300; }
.cr-border-orange { border-color: $cr-orange; }
.cr-border-orange-100 { border-color: $cr-orange-100; }
.cr-border-orange-200 { border-color: $cr-orange-200; }
.cr-border-orange-300 { border-color: $cr-orange-300; }
.cr-border-yellow { border-color: $cr-yellow; }
.cr-border-yellow-100 { border-color: $cr-yellow-100; }
.cr-border-yellow-200 { border-color: $cr-yellow-200; }
.cr-border-yellow-300 { border-color: $cr-yellow-300; }
.cr-border-gray { border-color: $cr-gray; }
.cr-border-gray-100 { border-color: $cr-gray-100; }
.cr-border-gray-200 { border-color: $cr-gray-200; }
.cr-border-gray-300 { border-color: $cr-gray-300; }
.cr-border-seal { border-color: $cr-seal; }
.cr-border-seal-100 { border-color: $cr-seal-100; }
.cr-border-seal-200 { border-color: $cr-seal-200; }
.cr-border-seal-300 { border-color: $cr-seal-300; }
"
`;

exports[`generate-utilities _radii.scss should output the right CSS 1`] = `
"@import '../index.scss';
.cr-radius-1 { border-radius: $cr-radius-1; }
.cr-radius-2 { border-radius: $cr-radius-2; }
.cr-radius-circle { border-radius: $cr-radius-circle; }
"
`;

exports[`generate-utilities _shadows.scss should output the right CSS 1`] = `
"@import '../index.scss';
.cr-shadow-1 { box-shadow: $cr-shadow-1; }
.cr-shadow-2 { box-shadow: $cr-shadow-2; }
.cr-shadow-3 { box-shadow: $cr-shadow-3; }
.cr-shadow-4 { box-shadow: $cr-shadow-4; }
.cr-shadow-5 { box-shadow: $cr-shadow-5; }
.cr-inner-shadow-1 { box-shadow: $cr-inner-shadow-1; }
.cr-inner-shadow-2 { box-shadow: $cr-inner-shadow-2; }
.cr-inner-shadow-n1 { box-shadow: $cr-inner-shadow-n1; }
.cr-border-shadow { box-shadow: $cr-border-shadow; }
"
`;

exports[`generate-utilities _spacing.scss should output the right CSS 1`] = `
"@import '../index.scss';
.cr-m-0 { margin: $cr-spacing-0; }
.cr-m-1 { margin: $cr-spacing-1; }
.cr-m-2 { margin: $cr-spacing-2; }
.cr-m-3 { margin: $cr-spacing-3; }
.cr-m-4 { margin: $cr-spacing-4; }
.cr-m-5 { margin: $cr-spacing-5; }
.cr-m-6 { margin: $cr-spacing-6; }
.cr-m-7 { margin: $cr-spacing-7; }
.cr-m-8 { margin: $cr-spacing-8; }
.cr-m-9 { margin: $cr-spacing-9; }
.cr-m-10 { margin: $cr-spacing-10; }
.cr-p-0 { padding: $cr-spacing-0; }
.cr-p-1 { padding: $cr-spacing-1; }
.cr-p-2 { padding: $cr-spacing-2; }
.cr-p-3 { padding: $cr-spacing-3; }
.cr-p-4 { padding: $cr-spacing-4; }
.cr-p-5 { padding: $cr-spacing-5; }
.cr-p-6 { padding: $cr-spacing-6; }
.cr-p-7 { padding: $cr-spacing-7; }
.cr-p-8 { padding: $cr-spacing-8; }
.cr-p-9 { padding: $cr-spacing-9; }
.cr-p-10 { padding: $cr-spacing-10; }
.cr-mt-0 { margin-top: $cr-spacing-0; }
.cr-mt-1 { margin-top: $cr-spacing-1; }
.cr-mt-2 { margin-top: $cr-spacing-2; }
.cr-mt-3 { margin-top: $cr-spacing-3; }
.cr-mt-4 { margin-top: $cr-spacing-4; }
.cr-mt-5 { margin-top: $cr-spacing-5; }
.cr-mt-6 { margin-top: $cr-spacing-6; }
.cr-mt-7 { margin-top: $cr-spacing-7; }
.cr-mt-8 { margin-top: $cr-spacing-8; }
.cr-mt-9 { margin-top: $cr-spacing-9; }
.cr-mt-10 { margin-top: $cr-spacing-10; }
.cr-ml-0 { margin-left: $cr-spacing-0; }
.cr-ml-1 { margin-left: $cr-spacing-1; }
.cr-ml-2 { margin-left: $cr-spacing-2; }
.cr-ml-3 { margin-left: $cr-spacing-3; }
.cr-ml-4 { margin-left: $cr-spacing-4; }
.cr-ml-5 { margin-left: $cr-spacing-5; }
.cr-ml-6 { margin-left: $cr-spacing-6; }
.cr-ml-7 { margin-left: $cr-spacing-7; }
.cr-ml-8 { margin-left: $cr-spacing-8; }
.cr-ml-9 { margin-left: $cr-spacing-9; }
.cr-ml-10 { margin-left: $cr-spacing-10; }
.cr-mr-0 { margin-right: $cr-spacing-0; }
.cr-mr-1 { margin-right: $cr-spacing-1; }
.cr-mr-2 { margin-right: $cr-spacing-2; }
.cr-mr-3 { margin-right: $cr-spacing-3; }
.cr-mr-4 { margin-right: $cr-spacing-4; }
.cr-mr-5 { margin-right: $cr-spacing-5; }
.cr-mr-6 { margin-right: $cr-spacing-6; }
.cr-mr-7 { margin-right: $cr-spacing-7; }
.cr-mr-8 { margin-right: $cr-spacing-8; }
.cr-mr-9 { margin-right: $cr-spacing-9; }
.cr-mr-10 { margin-right: $cr-spacing-10; }
.cr-mb-0 { margin-bottom: $cr-spacing-0; }
.cr-mb-1 { margin-bottom: $cr-spacing-1; }
.cr-mb-2 { margin-bottom: $cr-spacing-2; }
.cr-mb-3 { margin-bottom: $cr-spacing-3; }
.cr-mb-4 { margin-bottom: $cr-spacing-4; }
.cr-mb-5 { margin-bottom: $cr-spacing-5; }
.cr-mb-6 { margin-bottom: $cr-spacing-6; }
.cr-mb-7 { margin-bottom: $cr-spacing-7; }
.cr-mb-8 { margin-bottom: $cr-spacing-8; }
.cr-mb-9 { margin-bottom: $cr-spacing-9; }
.cr-mb-10 { margin-bottom: $cr-spacing-10; }
.cr-pt-0 { padding-top: $cr-spacing-0; }
.cr-pt-1 { padding-top: $cr-spacing-1; }
.cr-pt-2 { padding-top: $cr-spacing-2; }
.cr-pt-3 { padding-top: $cr-spacing-3; }
.cr-pt-4 { padding-top: $cr-spacing-4; }
.cr-pt-5 { padding-top: $cr-spacing-5; }
.cr-pt-6 { padding-top: $cr-spacing-6; }
.cr-pt-7 { padding-top: $cr-spacing-7; }
.cr-pt-8 { padding-top: $cr-spacing-8; }
.cr-pt-9 { padding-top: $cr-spacing-9; }
.cr-pt-10 { padding-top: $cr-spacing-10; }
.cr-pl-0 { padding-left: $cr-spacing-0; }
.cr-pl-1 { padding-left: $cr-spacing-1; }
.cr-pl-2 { padding-left: $cr-spacing-2; }
.cr-pl-3 { padding-left: $cr-spacing-3; }
.cr-pl-4 { padding-left: $cr-spacing-4; }
.cr-pl-5 { padding-left: $cr-spacing-5; }
.cr-pl-6 { padding-left: $cr-spacing-6; }
.cr-pl-7 { padding-left: $cr-spacing-7; }
.cr-pl-8 { padding-left: $cr-spacing-8; }
.cr-pl-9 { padding-left: $cr-spacing-9; }
.cr-pl-10 { padding-left: $cr-spacing-10; }
.cr-pr-0 { padding-right: $cr-spacing-0; }
.cr-pr-1 { padding-right: $cr-spacing-1; }
.cr-pr-2 { padding-right: $cr-spacing-2; }
.cr-pr-3 { padding-right: $cr-spacing-3; }
.cr-pr-4 { padding-right: $cr-spacing-4; }
.cr-pr-5 { padding-right: $cr-spacing-5; }
.cr-pr-6 { padding-right: $cr-spacing-6; }
.cr-pr-7 { padding-right: $cr-spacing-7; }
.cr-pr-8 { padding-right: $cr-spacing-8; }
.cr-pr-9 { padding-right: $cr-spacing-9; }
.cr-pr-10 { padding-right: $cr-spacing-10; }
.cr-pb-0 { padding-bottom: $cr-spacing-0; }
.cr-pb-1 { padding-bottom: $cr-spacing-1; }
.cr-pb-2 { padding-bottom: $cr-spacing-2; }
.cr-pb-3 { padding-bottom: $cr-spacing-3; }
.cr-pb-4 { padding-bottom: $cr-spacing-4; }
.cr-pb-5 { padding-bottom: $cr-spacing-5; }
.cr-pb-6 { padding-bottom: $cr-spacing-6; }
.cr-pb-7 { padding-bottom: $cr-spacing-7; }
.cr-pb-8 { padding-bottom: $cr-spacing-8; }
.cr-pb-9 { padding-bottom: $cr-spacing-9; }
.cr-pb-10 { padding-bottom: $cr-spacing-10; }

.cr-mv-0 {
  margin-top: $cr-spacing-0;
  margin-bottom: $cr-spacing-0;
}

.cr-mv-1 {
  margin-top: $cr-spacing-1;
  margin-bottom: $cr-spacing-1;
}

.cr-mv-2 {
  margin-top: $cr-spacing-2;
  margin-bottom: $cr-spacing-2;
}

.cr-mv-3 {
  margin-top: $cr-spacing-3;
  margin-bottom: $cr-spacing-3;
}

.cr-mv-4 {
  margin-top: $cr-spacing-4;
  margin-bottom: $cr-spacing-4;
}

.cr-mv-5 {
  margin-top: $cr-spacing-5;
  margin-bottom: $cr-spacing-5;
}

.cr-mv-6 {
  margin-top: $cr-spacing-6;
  margin-bottom: $cr-spacing-6;
}

.cr-mv-7 {
  margin-top: $cr-spacing-7;
  margin-bottom: $cr-spacing-7;
}

.cr-mv-8 {
  margin-top: $cr-spacing-8;
  margin-bottom: $cr-spacing-8;
}

.cr-mv-9 {
  margin-top: $cr-spacing-9;
  margin-bottom: $cr-spacing-9;
}

.cr-mv-10 {
  margin-top: $cr-spacing-10;
  margin-bottom: $cr-spacing-10;
}

.cr-pv-0 {
  padding-top: $cr-spacing-0;
  padding-bottom: $cr-spacing-0;
}

.cr-pv-1 {
  padding-top: $cr-spacing-1;
  padding-bottom: $cr-spacing-1;
}

.cr-pv-2 {
  padding-top: $cr-spacing-2;
  padding-bottom: $cr-spacing-2;
}

.cr-pv-3 {
  padding-top: $cr-spacing-3;
  padding-bottom: $cr-spacing-3;
}

.cr-pv-4 {
  padding-top: $cr-spacing-4;
  padding-bottom: $cr-spacing-4;
}

.cr-pv-5 {
  padding-top: $cr-spacing-5;
  padding-bottom: $cr-spacing-5;
}

.cr-pv-6 {
  padding-top: $cr-spacing-6;
  padding-bottom: $cr-spacing-6;
}

.cr-pv-7 {
  padding-top: $cr-spacing-7;
  padding-bottom: $cr-spacing-7;
}

.cr-pv-8 {
  padding-top: $cr-spacing-8;
  padding-bottom: $cr-spacing-8;
}

.cr-pv-9 {
  padding-top: $cr-spacing-9;
  padding-bottom: $cr-spacing-9;
}

.cr-pv-10 {
  padding-top: $cr-spacing-10;
  padding-bottom: $cr-spacing-10;
}

.cr-mh-0 {
  margin-left: $cr-spacing-0;
  margin-right: $cr-spacing-0;
}

.cr-mh-1 {
  margin-left: $cr-spacing-1;
  margin-right: $cr-spacing-1;
}

.cr-mh-2 {
  margin-left: $cr-spacing-2;
  margin-right: $cr-spacing-2;
}

.cr-mh-3 {
  margin-left: $cr-spacing-3;
  margin-right: $cr-spacing-3;
}

.cr-mh-4 {
  margin-left: $cr-spacing-4;
  margin-right: $cr-spacing-4;
}

.cr-mh-5 {
  margin-left: $cr-spacing-5;
  margin-right: $cr-spacing-5;
}

.cr-mh-6 {
  margin-left: $cr-spacing-6;
  margin-right: $cr-spacing-6;
}

.cr-mh-7 {
  margin-left: $cr-spacing-7;
  margin-right: $cr-spacing-7;
}

.cr-mh-8 {
  margin-left: $cr-spacing-8;
  margin-right: $cr-spacing-8;
}

.cr-mh-9 {
  margin-left: $cr-spacing-9;
  margin-right: $cr-spacing-9;
}

.cr-mh-10 {
  margin-left: $cr-spacing-10;
  margin-right: $cr-spacing-10;
}

.cr-ph-0 {
  padding-left: $cr-spacing-0;
  padding-right: $cr-spacing-0;
}

.cr-ph-1 {
  padding-left: $cr-spacing-1;
  padding-right: $cr-spacing-1;
}

.cr-ph-2 {
  padding-left: $cr-spacing-2;
  padding-right: $cr-spacing-2;
}

.cr-ph-3 {
  padding-left: $cr-spacing-3;
  padding-right: $cr-spacing-3;
}

.cr-ph-4 {
  padding-left: $cr-spacing-4;
  padding-right: $cr-spacing-4;
}

.cr-ph-5 {
  padding-left: $cr-spacing-5;
  padding-right: $cr-spacing-5;
}

.cr-ph-6 {
  padding-left: $cr-spacing-6;
  padding-right: $cr-spacing-6;
}

.cr-ph-7 {
  padding-left: $cr-spacing-7;
  padding-right: $cr-spacing-7;
}

.cr-ph-8 {
  padding-left: $cr-spacing-8;
  padding-right: $cr-spacing-8;
}

.cr-ph-9 {
  padding-left: $cr-spacing-9;
  padding-right: $cr-spacing-9;
}

.cr-ph-10 {
  padding-left: $cr-spacing-10;
  padding-right: $cr-spacing-10;
}
"
`;

exports[`generate-utilities _transition.scss should output the right CSS 1`] = `
"@import '../index.scss';
.cr-transition-duration-fast { transition-duration: $cr-transition-duration-fast; }
.cr-transition-duration-base { transition-duration: $cr-transition-duration-base; }
.cr-transition-duration-slow { transition-duration: $cr-transition-duration-slow; }
"
`;

exports[`generate-utilities _typography.scss should output the right CSS 1`] = `
"@import '../index.scss';
.cr-serif { font-family: $cr-serif; }
.cr-sans-serif { font-family: $cr-sans-serif; }
.cr-monospace { font-family: $cr-monospace; }

.cr-regular { font-weight: $cr-regular; }
.cr-bold { font-weight: $cr-bold; }

.cr-text-xxl { font-size: $cr-text-xxl; }
.cr-text-xl { font-size: $cr-text-xl; }
.cr-text-l { font-size: $cr-text-l; }
.cr-text { font-size: $cr-text; }
.cr-text-s { font-size: $cr-text-s; }
.cr-text-xs { font-size: $cr-text-xs; }
.cr-text-xxs { font-size: $cr-text-xxs; }

.cr-letter-spacing-normal { letter-spacing: $cr-letter-spacing-normal; }
.cr-letter-spacing-1 { letter-spacing: $cr-letter-spacing-1; }
.cr-letter-spacing-2 { letter-spacing: $cr-letter-spacing-2; }
.cr-letter-spacing-3 { letter-spacing: $cr-letter-spacing-3; }
"
`;

exports[`generate-utilities index.scss should output the right CSS 1`] = `
"@import '../index.scss';
@import 'typography.scss';
@import 'spacing.scss';
@import 'colors.scss';
@import 'shadows.scss';
@import 'radii.scss';
@import 'transition.scss';
"
`;
